quart_prometheus_logger.py

```diff
@@ -127,46 +140,57 @@ def _register_collectors(self):
                     "http_requests",
                     "Total number of requests",
                     ["method", "path", "status", *self._custom_label_names],
+                    registry=self._registry,
                 ),
                 Counter(
                     "http_requests_errors",
                     "Total number of error requests",
                     ["method", "path", "status", *self._custom_label_names],
+                    registry=self._registry,
                 ),
                 Histogram(
                     "http_request_duration_seconds",
                     "The amount of time spent handling requests",
                     ["path", *self._custom_label_names],
+                    registry=self._registry,
                 ),
                 Histogram(
                     "http_request_size_bytes",
                     "The size of requests",
                     ["path", *self._custom_label_names],
                     buckets=REQUEST_BUCKETS,
+                    registry=self._registry,
                 ),
                 Histogram(
                     "http_response_size_bytes",
                     "The size of responses",
                     ["path", *self._custom_label_names],
                     buckets=RESPONSE_BUCKETS,
+                    registry=self._registry,
                 ),
             )
         }
 
+    def _reset_metrics(self):
+        """Unregister all collectors from the registry."""
+        for _, collector in self._collectors.items():
+            self._registry.unregister(collector)
```

Comment:
Not a critique of this PR, just annoyed that they didn't go with `deregister` :disappointed:

Reply:
😞